pytorch_lightning/trainer/training_loop.py

```diff
@@ -463,6 +463,12 @@ def run_training_epoch(self):
             if batch_output.signal == -1:
                 break
 
+            # update non-plateau LR schedulers
+            # update epoch-interval ones only when we are at the end of training epoch
+            self.update_lr_schedulers('step', update_plateau_schedulers=False)
+            if self._num_training_batches_reached(is_last_batch):
+                self.update_lr_schedulers('epoch', update_plateau_schedulers=False)
```

Comment:
@carmocca To make sure epoch based lr_schedulers are updated before validation (and only once at the end of training epoch), I added this condition. Would this cover all cases or does it need to be refined more?

Reply:
Good point. I think that's okay